ksqldb-tools/src/main/java/io/confluent/ksql/tools/migrations/commands/ApplyMigrationCommand.java

```diff
@@ -298,6 +331,44 @@ private boolean applyMigration(
     return true;
   }
 
+  private KsqlObject getRow(
+      final List<FieldInfo> sourceFields,
+      final List<String> insertColumns,
+      final List<Expression> insertValues
+  ) {
+    final Map<String, Object> row = new HashMap<>();
+    if (insertColumns.size() > 0) {
+      if (insertColumns.size() != insertValues.size()) {
+        throw new MigrationException("mismatch");
+      }
+      for (int i = 0 ; i < insertColumns.size(); i++) {
+        final String columnName = insertColumns.get(i);
+        final List<FieldInfo> matchingFields = sourceFields.stream()
+            .filter(f -> f.name().equals(columnName)).collect(Collectors.toList());
```

Comment:
Originally the `sourceFields` were providing the type information, but that's unnecessary since the value expressions contain that information. I updated the PR so that we don't need to check anymore.  This also resolves the nested complex types problem.